python/deps/untypy/untypy/impl/protocol.py

```diff
@@ -232,10 +239,11 @@ def __setattr__(me, name, value):
     list_of_attr['__setattr__'] = __setattr__  # allow access of attributes
     name = f"{protocolchecker.proto.__name__}For{original.__name__}"
 
-    if type(original) == type and original.__flags__ & 0x0400:
+    if type(original) == type and original.__flags__ & 0x0400 and original not in [dict, list, set, tuple]:
```

Comment:
@CodeSteak Can you comment on this? Why is this necessary? I'm asking because when merging HEAD into #64, one test fails: when a string is used at type Sequence, then printing the value does not print to string but the empty string. Adding str to the blacklist fixes the problem. Nevertheless, I'd like to understand what's going on here because there might be other bugs waiting...

Reply:
Some "primitive" built-in types are implemented in C in the python interpreter. Inheriting from these types occasionally ignores overwritten methods and those from the C implementation are called. It seems this is not properly implemented in the interpreter. I believe #45 is also an interpreter bug.

Reply:
By using Metaclasses the `isinstance`-check (`instance(a, B)`) could be influenced, but only by the class that is checked against (`B`), not the class of the instance (`a`). So this not an option, as they can't be influenced.